server/service/src/vaccination/update/generate.rs

```diff
@@ -0,0 +1,134 @@
+use repository::{StockLine, VaccinationRow};
+use util::constants::REVERSE_PRESCRIPTION_REASON_ID;
+
+use crate::{
+    invoice::inventory_adjustment::{AdjustmentType, InsertInventoryAdjustment},
+    vaccination::generate::{
+        generate_create_prescription, get_dose_as_number_of_packs, CreatePrescription,
+    },
+    NullableUpdate,
+};
+
+use super::UpdateVaccination;
+
+pub struct GenerateInput {
+    pub patient_id: String,
+    pub existing_vaccination: VaccinationRow,
+    pub update_input: UpdateVaccination,
+    pub existing_stock_line: Option<StockLine>,
+    pub new_stock_line: Option<StockLine>,
+}
+
+pub struct GenerateResult {
+    pub vaccination: VaccinationRow,
+    pub create_inventory_adjustment: Option<InsertInventoryAdjustment>,
+    pub create_prescription: Option<CreatePrescription>,
+}
+
+pub fn generate(
```

Comment:
There's quite a bit of complexity in terms of changing to/from given, or changing the stock line etc... not sure if there is a clearer way to denote the forks/where logic is shared? Let me know any areas that seem particularly unclear 😢